poly1305/src/autodetect.rs

```diff
@@ -0,0 +1,59 @@
+//! Autodetection support for AVX2 CPU intrinsics on x86 CPUs, with fallback
+//! to the "soft" backend when it's unavailable.
+
+use crate::{backend, Block, Key, Tag};
+
+cpuid_bool::new!(avx2_cpuid, "avx2");
+
+#[derive(Clone)]
+pub struct State {
+    inner: Inner,
+}
+
+// TODO(tarcieri): switch to a `union` w\ `ManuallyDrop` when MSRV 1.49+
```

Comment:
`backend::avx2::State` is somewhat complicated, so I didn't want to temporarily slap `Copy` all over it.

I can try experimenting with that in a follow-up PR, but I'd prefer doing that in a way that's easy to revert when we can bump MSRV.

Reply:
Never mind, that was easier than expected!